notifications/listeners.py

```diff
@@ -0,0 +1,122 @@
+import logging
+
+from django.apps import apps
+
+from website.project.signals import contributor_added, project_created, node_deleted, contributor_removed
+from website.reviews import signals as reviews_signals
+
+logger = logging.getLogger(__name__)
+
+@project_created.connect
+def subscribe_creator(resource):
+    from osf.models import NotificationSubscription, NotificationType
+
+    from django.contrib.contenttypes.models import ContentType
+
+    if resource.is_collection or resource.is_deleted:
+        return None
+    user = resource.creator
+    if user.is_registered:
+        try:
+            NotificationSubscription.objects.get_or_create(
+                user=user,
+                notification_type=NotificationType.Type.USER_FILE_UPDATED.instance,
+                _is_digest=True,
+            )
+        except NotificationSubscription.MultipleObjectsReturned:
+            pass
+        NotificationSubscription.objects.get_or_create(
+            user=user,
+            notification_type=NotificationType.Type.FILE_UPDATED.instance,
+            object_id=resource.id,
+            content_type=ContentType.objects.get_for_model(resource),
+            _is_digest=True,
+        )
+
+@contributor_added.connect
+def subscribe_contributor(resource, contributor, auth=None, *args, **kwargs):
+    from django.contrib.contenttypes.models import ContentType
+    from osf.models import NotificationSubscription, NotificationType
+
+    from osf.models import Node
+    if isinstance(resource, Node):
+        if resource.is_collection or resource.is_deleted:
+            return None
+
+    if contributor.is_registered:
+        NotificationSubscription.objects.get_or_create(
+            user=contributor,
+            notification_type=NotificationType.Type.USER_FILE_UPDATED.instance,
+            object_id=contributor.id,
+            content_type=ContentType.objects.get_for_model(contributor),
+            _is_digest=True,
+        )
+        NotificationSubscription.objects.get_or_create(
+            user=contributor,
+            notification_type=NotificationType.Type.FILE_UPDATED.instance,
+            object_id=resource.id,
+            content_type=ContentType.objects.get_for_model(resource),
+            _is_digest=True,
+        )
```

Comment:
```suggestion
        NotificationSubscription.objects.bulk_create([
            NotificationSubscription(
                user=contributor,
                notification_type=NotificationType.Type.USER_FILE_UPDATED.instance,
                object_id=contributor.id,
                content_type=ContentType.objects.get_for_model(contributor),
                _is_digest=True,
            ),
            NotificationSubscription(
                user=contributor,
                notification_type=NotificationType.Type.FILE_UPDATED.instance,
                object_id=resource.id,
                content_type=ContentType.objects.get_for_model(resource),
                _is_digest=True,
            )
        ], ignore_conflicts=True)
```

May be simplified into one db query